api/v1alpha1/envoyproxy_metric_types.go

```diff
@@ -29,6 +29,9 @@ type ProxyMetrics struct {
 
 	// EnableVirtualHostStats enables envoy stat metrics for virtual hosts.
 	EnableVirtualHostStats bool `json:"enableVirtualHostStats,omitempty"`
+
+	// EnablePerEndpointStats enables envoy stats metrics for peer endpoint
+	EnablePerEndpointStats bool `json:"enablePerEndpointStats,omitempty"`
```

Comment:
In kubernetes world, endpoints may change frequently, would that cause high memory usage?
If so, I suggest EG should highlight that case.

Reply:
When the upstream cluster reaches a certain order of magnitude, this "button" becomes a black box. We may need to add comments to the field to inform users to be cautious about enabling this feature.

Reply:
yeah, EG definitely should warn/highlight it.

Reply:
done